### PR TITLE
Unify handling of Samples

### PR DESCRIPTION
**Pull request by dblenkus**
Requires: https://github.com/genialis/resolwe/pull/150

### CODE REVIEW

**Comment by dblenkus** [comment]
I've fixed linter errors. Tests will pass when all requirements will be accepted.

File: resolwe_bio/signals.py [hunk 1]
```diff
@@ -56,4 +56,21 @@ def add_post_save_handler(sender, instance, **kwargs):
...
+            #      object is added to collections.
+            # for collection in Collection.objects.filter(data=instance.pk):
+            #     sample.collections.add(collection)
+
         sample.data.add(instance)
+
+
+@receiver(pre_delete, sender=Data)
+def add_pre_delete_handler(sender, instance, **kwargs):
+    """Delete Sample when last Data object is deleted."""
```
Comment (mstajdohar) [inline resolwe_bio/signals.py]: Did you think about deleting a Sample when the last Data object is removed?

Comment (dblenkus) [inline resolwe_bio/signals.py]: Good point. I've added this.

Comment (mstajdohar) [inline resolwe_bio/signals.py]: Well, I am still thinking if we want this or not. But in any case, we have to either delete the Sample in both cases or not delete it at all.

**Pull request merged**